pkg/datapath/tables/device.go

```diff
@@ -97,6 +97,20 @@ type Device struct {
 	NotSelectedReason string // Reason why this device was not selected
 }
 
+func (d *Device) DeepEqual(other *Device) bool {
```

Comment:
Add a comment to "Device" struct that if you add/modify fields this function needs to be updated.

Reply:
I was able to autogenerate the DeepEqual for the Device which is the largest, the DeviceAddress still needs a manual deep equal because the generation tooling doesn't understand that you can use == on `netip.Addr`.